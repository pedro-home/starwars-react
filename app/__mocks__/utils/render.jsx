import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import configureStore from 'store';
import Refs from 'contexts/Refs';
import { DEFAULT_REFS } from 'contexts/Refs/constants';

export function renderWithRedux(component, initialState = {}) {
  const store = configureStore(initialState);
  return {
    ...render(<Provider store={store}>{component}</Provider>),
    store,
  };
}

export function renderWithApp(component, initialState = {}, refsValue = DEFAULT_REFS) {
  return renderWithRedux(<Refs.Provider value={refsValue}>{component}</Refs.Provider>, initialState);
}
