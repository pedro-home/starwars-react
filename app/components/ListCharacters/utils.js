import femaleLogo from './assets/female.jpg';
import maleLogo from './assets/male.jpg';
import hermaphroditeLogo from './assets/hermaphrodite.jpg';
import droidLogo from './assets/droid.jpg';
import { OFFSET_SCROLL } from './constants';

export const getCharacterImage = name => {
  let logo;
  switch (name) {
    case 'female':
      logo = femaleLogo;
      break;
    case 'male':
      logo = maleLogo;
      break;
    case 'hermaphrodite':
      logo = hermaphroditeLogo;
      break;
    default:
      logo = droidLogo;
  }

  return logo;
};

export const hasEndedScroll = (offset = OFFSET_SCROLL) => {
  return window.scrollY + window.innerHeight >= document.body.scrollHeight - offset;
};
