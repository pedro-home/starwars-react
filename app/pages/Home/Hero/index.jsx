import React from 'react';
import { FlexboxGrid, Col } from 'rsuite';
import Search from 'containers/Search';
import logo from './assets/logo.png';

const Section = () => {
  return (
    <section className="hero">
      <div className="center-container">
        <FlexboxGrid justify="center">
          <FlexboxGrid.Item componentClass={Col} md={12}>
            <img className="logo" src={logo} />
            <Search />
          </FlexboxGrid.Item>
        </FlexboxGrid>
      </div>
    </section>
  );
};

export default Section;
