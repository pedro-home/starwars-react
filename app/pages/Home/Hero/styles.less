.hero {
  position: relative;
  height: 100vh;

  .container-center {
    max-width: 50rem;
    top: 40%;
  }
}

.logo {
  width: 100%;
  display: block;
  margin: 0 auto 5rem auto;

  @media (min-width: @screen-desktop) {
    padding: 0 5rem;
  }
}
