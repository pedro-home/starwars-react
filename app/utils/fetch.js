function fetch(url, options) {
  return window.fetch(url, options).then(response => {
    if (response.ok) {
      let body;
      const contentType = response.headers.get('content-type');
      if (contentType && contentType.indexOf('application/json') !== -1) {
        body = response.json();
      } else {
        body = response.text();
      }

      return body;
    }

    throw new Error(`${response.status}: ${response.statusText}`);
  });
}

export default fetch;
