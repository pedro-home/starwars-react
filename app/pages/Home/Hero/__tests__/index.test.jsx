import React from 'react';
import { renderWithApp } from '__mocks__/utils/render';

import Hero from '../index';

describe('Given <Hero />', () => {
  let component;

  describe('When initializes', () => {
    beforeEach(() => {
      const { container } = renderWithApp(<Hero />);
      component = container;
    });

    it('Then should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });

    it('And should render a center container', () => {
      expect(component.querySelector('.container-center')).toBeTruthy();
    });

    it('And should render a logo', () => {
      expect(component.querySelector('.logo')).toBeTruthy();
    });

    it('And should render a search', () => {
      expect(component.querySelector('.search')).toBeTruthy();
    });
  });
});
