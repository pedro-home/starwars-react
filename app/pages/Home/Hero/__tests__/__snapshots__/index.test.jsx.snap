// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <Hero /> When initializes Then should match the snapshot 1`] = `
<div>
  <section
    class="section hero"
  >
    <div
      class="center-container"
    >
      <img
        class="logo"
        src="logo.png"
      />
      <div
        class="rs-input-group search rs-input-group-inside"
      >
        <input
          class="rs-input"
          placeholder="Search databank..."
          type="text"
          value=""
        />
        <a
          class="rs-btn rs-btn-default rs-input-group-btn"
        >
          <i
            class="rs-icon rs-icon-search"
          />
          <span
            class="rs-ripple-pond"
          >
            <span
              class="rs-ripple"
            />
          </span>
        </a>
      </div>
    </div>
  </section>
</div>
`;
